csi/controllerserver.py

```diff
@@ -255,6 +258,118 @@ def ControllerGetCapabilities(self, request, context):
                     "rpc": {
                         "type": capability_type("LIST_VOLUMES")
                     }
+                },
+                {
+                    "rpc": {
+                        "type": capability_type("EXPAND_VOLUME")
+                    }
                 }
             ]
         )
+
+    def ControllerExpandVolume(self, request, context):
+        """
+        Controller plugin RPC call implementation of EXPAND_VOLUME
+        """
+
+        start_time = time.time()
+        expansion_requested_pvsize = request.capacity_range.required_bytes
+
+        # Get existing volume size
+        existing_volume = search_volume(request.volume_id)
+        existing_pvsize = existing_volume.size
+
+        # pvtype = PV_TYPE_SUBVOL
+
+        # if KADALU_VERSION in ["0.5.0", "0.4.0", "0.3.0"]:
+        #     for vol_capability in request.volume_capabilities:
+        #         # using getattr to avoid Pylint error
+        #         single_node_writer = getattr(csi_pb2.VolumeCapability.AccessMode,
+        #                                     "SINGLE_NODE_WRITER")
+
+        #         if vol_capability.access_mode.mode == single_node_writer:
+        #             pvtype = PV_TYPE_VIRTBLOCK
+
+        pvtype = PV_TYPE_SUBVOL
+
+        single_node_writer = getattr(csi_pb2.VolumeCapability.AccessMode,
+                                             "SINGLE_NODE_WRITER")
+
+        if request.volume_capability.AccessMode == single_node_writer:
+            pvtype = PV_TYPE_VIRTBLOCK)
+
+        logging.debug(logf(
+            "Found PV type",
+            pvtype=pvtype,
+            capability=request.volume_capability
+        ))
+
+        # Add everything from parameter as filter item
+        filters = {}
+        for pkey, pvalue in request.parameters.items():
+            filters[pkey] = pvalue
+
+        logging.debug(logf(
+            "Filters applied to choose storage",
+            **filters
+        ))
+
+        # UID is stored at the time of installation in configmap.
+        uid = None
+        with open(os.path.join(VOLINFO_DIR, "uid")) as uid_file:
+            uid = uid_file.read()
+
+        host_volumes = get_pv_hosting_volumes(filters)
```

Comment:
What is the right way of getting the host_vol.. Since 'filters' is not available through ControllerExpandVolume request.parameters is not supported

Reply:
dump what you get in `request` and `context`... that should give you enough hints.

Reply:
I got it through search_volume(request.volume_id)